### ISSUE TITLE
`never` properties should not require the properties to be present

### ISSUE THREAD

**Issue by yuhr** [microsoft/TypeScript#43954]
# Bug Report

### 🔎 Search Terms

- never
- properties
- "missing in type"

### 🕗 Version & Regression Information

- Tested in nightly, 4.2.3, and 3.3.3

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.2.3#code/MYewdgzgLgBAHgLhgbxgMxCJYCmA3HAJxgF8YBeFEgbgFgAoAekZgAVCQAHIqATxgDkGEAJgBLCDAC2EiGLABzcWBh9ug5CVEAjAK6xCOAI66xhgCbLVvdQNTDs+ItVICAdAAoATAHYALACMAJRAA)

### 💻 Code

```ts
const x: { foo: never } = {};
// Property 'foo' is missing in type '{}' but required in type '{ foo: never; }'.(2741)
```

### 🙁 Actual behavior

- Object literals which are expected to have `never` properties require initialization for those impossible properties 

### 🙂 Expected behavior

- `never` properties should disallow any initialization

I see there existed a similar issue #16762, but I think this bug is not related to "optional" concept. I believe the current behavior is definitely weird.

**Comment by jcalz** [comment]
Well, it's not a bug... that's how object types and `never` work.  If you have an object of type `{bar: string}`, then it must have a `bar` property of type `string`.  If you have an object of type `{foo: never}`, then it must have a `foo` property of type `never`.  Since one cannot get one's hands on a value of type `never` to place into that object, it implies that values of type `{foo: never}` are similarly elusive.  But this is a feature of `never`, not a bug.  

If you want to be able to leave out a property, then you need to make that property optional (here's that "optional" concept).  So my suggestion would be to use something like `{foo?: never}` to mean "an object which cannot have a `foo` property" instead of `{foo: never}` which means "an object which must have an impossible `foo` property".

I can imagine changing this into a *suggestion* that properties of type `never` be interpreted the way you want, but then this suggestion would be a breaking change, and I don't know that anyone would think it's worth breaking existing code for it.

**Comment by RyanCavanaugh** [comment]
A type with a property of type `never` has zero legal values; `{ }` is not one of them.

**Comment by yuhr** [comment]
> Since one cannot get one's hands on a value of type `never` to place into that object, it implies that values of type `{foo: never}` are similarly elusive. But this is a feature of `never`, not a bug.

> A type with a property of type `never` has zero legal values; `{ }` is not one of them.

Okay, that sounds. Note that the suggested type `{ foo?: never }` results in `{ foo?: undefined }` which is not the desired type at this point, but it should be eventually resolved by #43947 (I've tested with the [insider package](https://github.com/microsoft/TypeScript/pull/43947#issuecomment-832150934) and it worked), so once it has been merged successfully, I won't suggest any breaking change about this.

Thank you all! I'd open a suggestion anew in case the PR was declined.
